gravatar/src/main/java/com/gravatar/services/ProfileService.kt

```diff
@@ -122,35 +123,81 @@ public class ProfileService(okHttpClient: OkHttpClient? = null) {
                 Result.Failure(errorTypeFromHttpCode(response.code()))
             }
         }
+
+    /**
+     * Fetches a Gravatar profile for the given hash or username.
+     * This method will catch any exception that occurs during the execution and return it as a [Result.Failure].
+     *
+     * @param hashOrUsername The hash or username to fetch the profile for
+     * @return The fetched profile
+     */
+    public suspend fun retrieveCatching(hashOrUsername: String): Result<Profile, ErrorType> = runCatchingService {
+        retrieve(hashOrUsername)
+    }
+
+    /**
+     * Fetches a Gravatar profile for the given email address.
+     * This method throws any exception that occurs during the execution.
+     *
+     * @param email The email address to fetch the profile for
+     * @return The fetched profiles
+     */
+    public suspend fun retrieve(email: Email): Result<Profile, ErrorType> {
```

Comment:
```suggestion
    public suspend fun retrieve(email: Email): Profile {
```

I think that we should return only when the operation is successful thus we don't need the `Result` return type. All errors should be thrown. Otherwise, users will have to implement two error handling for this method. Thoughts on this?

Reply:
> users will have to implement two error handling for this method

Yeah, that's a good point. We'll need to figure out what exceptions to throw and whether we need to declare our own exceptions for each of the current ErrorTypes. Anyway, I think it's a great suggestion. I'll iterate over it. Thanks @AdamGrzybkowski .

Reply:
I've pushed some changes. It should be ready for another round. 🙏🏼